cmd/podman/pods/create.go

```diff
@@ -134,6 +134,12 @@ func create(cmd *cobra.Command, args []string) error {
 		imageName = infraImage
 	}
 	img := imageName
+
+	if !cmd.Flag("infra").Changed && (cmd.Flag("share").Changed && (share == "none" || share == "")) {
```

Comment:
I don't think you have to check if shared is changed. Just checking the string should be good enough.